### ISSUE TITLE
Unicode characters in fish_prompt.fish can corrupt editing

### ISSUE THREAD

**Issue by h-b** [fish-shell/fish-shell#8071]
- fish version: 3.2.2
- Operating System: PureOS (Byzantium)
- Terminal: Issue is reproducible with Konsole, MATE Terminal and XTerm
- Issue is not reproducible after executing  `sh -c 'env HOME=$(mktemp -d) fish'`

This issue is caused by my file ~/.config/fish/functions/fish_prompt.fish with the following content:
```
function fish_prompt --description 'Write out the prompt'
	set laststatus $status

    set_color -b black
    printf '%s%s%s%s%s%s%s%s%s%s%s%s%s' (set_color -o white) '❰' (set_color green) (date +%H:%M:%S) (set_color white) '❙' (set_color yellow) (echo $PWD | sed -e "s|^$HOME|~|") (set_color white) $git_info (set_color white) '❱' (set_color white)
    if test $laststatus -eq 0
        printf "%s✔%s≻%s " (set_color -o green) (set_color white) (set_color normal)
    else
        printf "%s✘%s≻%s " (set_color -o red) (set_color white) (set_color normal)
    end
end
```

This causes wrong character display when using the tab key or as soon as keyboard input reaches the right edge of the terminal window. It happens rarely, but those cases that do happen are reproducible.

For example, when typing `md<tab>5` fish displays characters `md5su5`. After completing to type `md5sum`, fish displays `md5md5sum`. See https://asciinema.org/a/h1qQNu5L7PccRFalf28C9JUoj

This not only affects display, but corrupts the cursor position. For example, after moving the cursor to the start of the line, it is not displayed there, but a few character to the right.

My point is, above `fish_prompt.fish` should not cause such issues, as it only modifies the prompt. It seems to be related to the use of special unicode characters - when I replace those, the issue does not occur any more. I am opening an issue for this, as it caused me lots of time to find that my modified prompt causes it, maybe others have similar problems. The issue happens more often with long lines that need to be wrapped.

**Comment by faho** [comment]
https://fishshell.com/docs/current/faq.html#i-m-getting-weird-graphical-glitches-a-staircase-effect-ghost-characters

This particular case should be fixed by #8011.

**Comment by h-b** [comment]
I just tested with 3.2.2-413-g8787179a4 - yes it works now, thank you.
